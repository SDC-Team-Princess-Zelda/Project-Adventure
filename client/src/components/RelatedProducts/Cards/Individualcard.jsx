/* eslint-disable react/prop-types */
import React from 'react';
import styled from 'styled-components';
import ProductImg from './ProductImg';
import Star from '../../../Star';

// receives array [product information, thumbnail url]
function Individualcard({ product }) {
  const starRating = product[2].avgRating;
  return (
    <div>
      <IndCard>
        <ThumbnailImage>
          <ProductImg image={product[1]} product={product[0].id} />

        </ThumbnailImage>
        <CardText>
          <CategoryText>
            <div>{product[0].category}</div>
          </CategoryText>
          <p>{product[0].name}</p>
          <p>
            $
            {product[0].default_price}
          </p>
          <div>
            <Star score={starRating} />
            {' '}
          </div>
        </CardText>
      </IndCard>
    </div>
  );
}

const IndCard = styled.div`
  display: table-cell, relative;
  position: relative;
  border-style: solid;
  border-width: 2px;
  width: fit-content;
  height: fit-content;
  margin-right: 30px;
  margin-bottom: 5px;
  border-radius: 5px;
  padding: 0 15px 0 15px;
  &:hover {
    box-shadow: 0 8px 16px 0;
  }
`;

const CategoryText = styled.div`
  font-style: italic;
`;
const CardText = styled.div`
  padding-left: 1px;
`;

const ThumbnailImage = styled.div`
  margin-top: 5px;
`;

export {
  Individualcard,
  IndCard,
};
