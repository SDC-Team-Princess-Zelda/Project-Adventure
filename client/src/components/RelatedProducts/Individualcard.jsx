import React, {
  createContext, useContext, useEffect, useState,
} from 'react';
import styled from 'styled-components';
import axios from 'axios';
import { useRelated } from './RelatedContext';

// should retrieve all related products from initial product page
// map each individual card with all the information

export default function Individualcard({ product }) {
  return (
    <div>
      <CardText>
        <p>{product.category}</p>
        <p>{product.name}</p>
        <p>
          $
          {product.default_price}
        </p>
        <p>CSS rating with stars</p>
      </CardText>
    </div>
  );
}

const IndCard = styled.div`
  border-style: solid;
  border-width: 2px;
  width: fit-content;
  margin-right: 30px;
  margin-bottom: 5px;
  border-radius: 5px;
  padding: 0 15px 0 15px;
  &:hover {
    box-shadow: 0 8px 16px 0;
  }
`;

const CardText = styled.ul`
  align-item: left;
`;

export {
  Individualcard,
  IndCard,
};
