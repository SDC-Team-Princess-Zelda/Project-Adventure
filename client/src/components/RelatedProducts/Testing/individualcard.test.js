import Individualcard from '../Individualcard';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<Individualcard />).toJSON();
  expect(tree).toMatchSnapshot();
});

