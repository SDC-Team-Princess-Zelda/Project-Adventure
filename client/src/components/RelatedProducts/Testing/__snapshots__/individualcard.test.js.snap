// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <p
    className="Individualcard__CardText-gDdxCt bzZqqO"
  >
    <p>
      Accessories
    </p>
    <p>
      Bright Future Sunglasses
    </p>
    <p>
      $
      69.00
    </p>
    <span>
      CSS rating with stars
    </span>
  </p>
</div>
`;
